corehq/apps/reports/models.py

```diff
@@ -358,17 +362,17 @@ def url_kwargs(self):
         if self.subreport_slug:
             kwargs['subreport_slug'] = self.subreport_slug
 
-        return kwargs
+        return immutabledict(kwargs)
 
     @property
     @memoized
     def view_kwargs(self):
-        kwargs = self.url_kwargs
+        kwargs = dict(**self.url_kwargs)
 
         if not self.is_configurable_report:
             kwargs['permissions_check'] = self._dispatcher.permissions_check
 
-        return kwargs
+        return immutabledict(kwargs)
```

Comment:
Yay, `immutabledict`!  FYI, I think this can be written like so:
```python
if not self.is_configurable_report:
    return self.url_kwargs.union({'permissions_check': self._dispatcher.permissions_check})
 
return self.url_kwargs
```